### PR TITLE
Change StokesDrift example in docs to a wavepacket moving with group velocity

### LINKED ISSUE #3422: Three-dimensional `StokesDrift` docstring example may be misleading
The docstring example for three-dimensional Stokes drift can be improved:

https://github.com/CliMA/Oceananigans.jl/blob/1c2a6f8752b6425bf30d856f8ba0aa681c0ab818/src/StokesDrifts.jl#L219-L226

The `v`-component (for example) is associated with a Stokes drift `vS` of the form:

```julia
vS = p.Uˢ * exp(z / p.h) * cos(p.k * x) * sin(t)
```

For this to represent a real surface wave field, then it would have to be sinusoidally oscillating in time (hard to imagine). Also, the envelope of the surface drift is modulated in space by `cos(k * x)`. Unfortunately, this is quite confusing because `k` is typically used to represent the wavelength of the _surface wave field_ (not the envelope of the surface waves). For example, for a monochromatic wave field, `h = 2k` is the vertical scale of the Stokes drift. So in this example, `k` has a different meaning (governing the wave field envelope) then it normally does (the wavenumber of the underlying wave field).

A better docstring example might represent a wavepacket that is translating at the group velocity, as done in a validation case:

https://github.com/CliMA/Oceananigans.jl/blob/1c2a6f8752b6425bf30d856f8ba0aa681c0ab818/validation/stokes_drift/surface_wave_quasi_geostrophic_flow.jl#L5-L51

### PR DESCRIPTION
**Pull request by navidcy**
Closes #3422

cc @BrodiePearson @LeeAra0

### CODE REVIEW

**Comment by BrodiePearson** [comment]
Looks good to me. My only question is about the derivation of ``wˢ``, which results from a vertical integration of the analytical functions. Currently this integral is performed under the assumption that ``wˢ`` is zero at large depths, which I clarified in the code:
https://github.com/CliMA/Oceananigans.jl/blob/06f6ac746742d08c78b3cfed0bd40f7df4392819/src/StokesDrifts.jl#L213-L215

**Is this the correct assumption here?** It effectively means that ``wˢ`` and its gradients are a maximum at the surface under a depth-decaying Stokes drift, so any horizontal convergence/divergence of Stokes drift manifests a local change in sea surface height:
https://github.com/CliMA/Oceananigans.jl/blob/06f6ac746742d08c78b3cfed0bd40f7df4392819/src/StokesDrifts.jl#L248-L250

An alternative would be to set ``wˢ`` to zero at the surface, which would cause ``wˢ`` and its horizontal gradients to reach a maximum at large depths, as the occurrences of ``ûˢ(z)`` above would become ``( ûˢ(z) - ûˢ(0) )``. Essentially, horizontal convergence/divergence would manifest as downwelling/upwelling (rather than SSH changes in the current case). I feel like this approach makes sense for LES setups, but it could cause problems for simulations of the full water column where you don't want direct Stokes drift effects deep in the ocean interior.

I'm happy to go forward with this example as is, but the above would be a useful question to answer in the future.

**Comment by navidcy** [comment]
Thanks for the addition!

**Comment by navidcy** [comment]
@BrodiePearson if it looks OK can you approve the PR?

File: src/StokesDrifts.jl [hunk 1]
```diff
@@ -204,39 +204,61 @@ Similarly, on a grid with `topology = (Periodic, Periodic, Bounded)` and `parame
...
-@inline ∂x_vˢ(x, y, z, t, p) = - p.Uˢ * exp(z / p.h) * p.k * sin(p.k * x) * sin(t)
-@inline ∂z_uˢ(x, y, z, t, p) =   p.Uˢ * exp(z / p.h) / p.h * cos(p.k * x) * sin(t)
-@inline ∂z_vˢ(x, y, z, t, p) =   p.Uˢ * exp(z / p.h) / p.h * cos(p.k * x) * sin(t)
+@inline A(ξ, η) = exp(- (ξ^2 + η^2) / 2δ^2)
 
-stokes_drift_parameters = (Uˢ = 0.01, h = 20, k = 2π * 1e-2)
-stokes_drift = StokesDrift(; ∂x_vˢ, ∂z_uˢ, ∂z_vˢ, ∂t_uˢ, ∂t_vˢ, parameters=stokes_drift_parameters)
+@inline ∂ξ_A(ξ, η) = - ξ / δ^2 * A(ξ, η)
+@inline ∂η_A(ξ, η) = - η / δ^2 * A(ξ, η)
+@inline ∂η_∂ξ_A(ξ, η) = - η * ξ / δ^4 * A(ξ, η)
```
[the remark this answers is not in the document]
Comment (navidcy) [reply src/StokesDrifts.jl]: I fixed it.

**Review by glwagner — APPROVED** [review]

**Pull request merged**